pkg/scheduler/eventhandlers.go

```diff
@@ -160,6 +160,16 @@ func (sched *Scheduler) updatePodInSchedulingQueue(oldObj, newObj interface{}) {
 
 	logger.V(4).Info("Update event for unscheduled pod", "pod", klog.KObj(newPod))
 	sched.SchedulingQueue.Update(logger, oldPod, newPod)
+	if hasNominatedNodeNameChanged(oldPod, newPod) {
+		// Nominated node changed in pod, so we need to treat it as if the pod was deleted,
+		// because the lower or equal priority pods treat such a pod as if it was assigned.
+		sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(logger, framework.EventAssignedPodDelete, oldPod, nil, nil)
```

Comment:
Can we somehow make it effective only for the unschedulable pods that is lower/equal priority than this changed pod?

Reply:
I was thinking about it, but I was comparing it with preemption mechanism only. Let me think about it laud.

IIUC not all lower priority pods can be preempted, so it's possible that some higher priority pod was also blocked on them. This is why completion of some pod triggers reevaluation of all pods, not only lower and equal priority ones.

On the other hand, it seems like lower priority nominated pods are removed unconditionally in attempt to fit higher priority one. I also assume that in case intra pod affinity of a lower priority nominated pod was blocking scheduling, such pod would be preempted (nomination cleared), so effectively even here, it cannot block higher priority one.

So considering how lower priority nominated pods are cleared unconditionally, it seems we can indeed reschedule only lower or equal priority pods.

Reply:
WDYT about just defining `MoveLEPriorityToActiveOrBackoffQueue`?

Reply:
If we can achieve lower/equal priority pod filtering with `preCheck` simply, I would just use `MoveAllToActiveOrBackoffQueue`, but if we need something complicated, then a new function would make sense.

Reply:
I like the idea with new method suited precisely for lower priority pods moving. I would limit usage of preCheck outside the scheduling queue, as it's risky. PreCheck can be used in this new method obviously, calling internal `moveAllToActiveOrBackoffQueue`.

Reply:
I don't regard my opinion above that strong, either is fine actually. Just go with what you'd like